susemanager-utils/susemanager-sls/salt/bootstrap/init.sls

```diff
@@ -26,14 +26,31 @@ mgr_server_localhost_alias_absent:
 {%- else %}
 {% set bootstrap_repo_url = 'https://' ~ salt['pillar.get']('mgr_server') ~ '/pub/repositories/' ~ os_base ~ '/' ~ grains['osrelease'] ~ '/0/bootstrap/' %}
 {%- endif %}
+
 {%- elif grains['os_family'] == 'RedHat' %}
-{%- if salt['file.file_exists' ]('/etc/oracle-release') %}
+{%- if salt['file.file_exists']('/etc/oracle-release') %}
 {% set bootstrap_repo_url = 'https://' ~ salt['pillar.get']('mgr_server') ~ '/pub/repositories/oracle/' ~ grains['osmajorrelease'] ~ '/bootstrap/' %}
-{%- elif salt['file.file_exists' ]('/etc/redhat-release') %}
+
+{%- elif salt['file.file_exists']('/usr/share/doc/sles_es-release') %}
+{% set bootstrap_repo_url = 'https://' ~ salt['pillar.get']('mgr_server') ~ '/pub/repositories/res/' ~ grains['osmajorrelease'] ~ '/bootstrap/' %}
+
+{%- elif salt['file.file_exists']('/etc/centos-release') %}
+{# We try CentOS bootstrap repository first. Failback to RES #}
+{% set bootstrap_repo_url = 'https://' ~ salt['pillar.get']('mgr_server') ~ '/pub/repositories/centos/' ~ grains['osmajorrelease'] ~ '/bootstrap/' %}
+{% set bootstrap_repo_request = salt['http.query'](bootstrap_repo_url + 'repodata/repomd.xml', status=True, verify_ssl=False) %}
+{%- if bootstrap_repo_request['status'] == 901 %}
+{{ raise(bootstrap_repo_request['error']) }}
+{%- elif not (0 < bootstrap_repo_request['status'] < 300) %}
```

Comment:
This is so you can follow redirects? But if so, how do you new the target of a redirect is valid?

Reply:
redirects are 301 or 302. So this is mainly checking 200 codes which are success codes

Reply:
Same as above. I'm reusing the same code/logic we already have in this SLS file. We do the same with whatever bootstrap repo we set.

Reply:
$me stupid: `< 300`